viewcontroller/viewcontroller.go

```diff
@@ -106,6 +107,17 @@ func NewRolloutViewController(rolloutId string, noColor bool) *RolloutViewContro
 
 func (c *RolloutViewController) GetRollout() (rollout.DetailedRollout, error) {
 	detailedRollout := rollout.DetailedRollout{}
+
+	if c.previousRollout == nil {
+		strategy, err := oceancd.GetStrategy(c.rolloutId)
+		if err != nil {
+			return detailedRollout, err
+		}
+		detailedRollout.Definition.Strategy = strategy
+	} else {
+		detailedRollout.Definition.Strategy = c.previousRollout.Definition.Strategy
```

Comment:
why do you take the strategy from the previous rollout? what if the user changed the strategy for the new rollout?

Reply:
here the previous rollout is the one fetched during the previous iteration. This is done to avoid extra API calls while using the `-w` argument. So we fetch the strategy for the current rollout only once.

Reply:
thanks for the explanation